### DISCUSSION TITLE
usb: usb don't work after switch from Zephyr 2.7.3 to 3.2.99 on i.MX RT1020

### DISCUSSION THREAD

**Discussion by MikhailSiomin** [zephyrproject-rtos/zephyr#53064] [Q&A] [answered]
I have a proprietary board with a processor i.MX RT1020. After switch from Zephyr 2.7.3 to 3.2.99, USB-OTG1 CDC/ACM does not work.
Also the cdc_acm example does not work on this board.

Maybe I need to add some config?

**Environment:**
- OS: Windows 10 Pro (21H2)
- Toolchain Zephyr SDK 0.15.1
- Zephyr version 3.2.99 (commit a9ed11ab8dcd6b7f6f677036694f9b9fa5dba74d)

**Comment by desowin** [comment]
What do you mean by "does not work"? Is the device enumerating at all? Does the device show in Device Manager? Are there any URBs captured when capturing newly connected devices with Wireshark/USBPcap?

**Comment by DerekSnell** [comment]
Hi @MikhailSiomin ,
I tested the sample `samples/subsys/usb/cdc_acm` using commit a9ed11ab8dcd6b7f6f677036694f9b9fa5dba74d on the MIMXRT1020-EVK board, and this sample works fine for me, see details below.  I also tested with the latest commit in main (2d3365200c9bb0ebdd0c3dc95f0ac04ca8ff046d), and that commit works fine for me too.

Do you have a MIMXRT1020-EVK board to compare with?  Are you able to review the differences of board files of your custom board with this board?  Since the in-tree board has no issue with this sample, I am going to move your post here to a discussion, and we can continue to investigate why it does not work on your board.  

Below is the debug console output for me.  The RT1020 enumerates in Windows as a VCOM port, and connecting a terminal to that port, it echoes back the characters as expected.
```
*** Booting Zephyr OS build zephyr-v3.2.0-2387-ga9ed11ab8dcd ***
[00:00:00.000,396] <inf> cdc_acm_echo: Wait for DTR
[00:00:05.305,023] <inf> usb_cdc_acm: Device configured
[00:00:20.469,970] <inf> cdc_acm_echo: DTR set
[00:00:21.474,060] <inf> cdc_acm_echo: Baudrate detected: 115200
```

Best regards

**Comment by MikhailSiomin** [comment, marked as answer]
Hi, @DerekSnell.
I discovered the following:
- a usb device built for the rt1020 in Zephyr version `3.2.99` `works` with Windows 10 version 21H2 build `19044.2251` when connected directly to a usb host on the motherboard or a usb host via a usb hub.
- a usb device built for the rt1020 in Zephyr version `3.2.99` `works` with Windows 10 version 22H2 build `19045.2364` `only` when connected `directly` to a usb host on the `motherboard`.
- a usb device built for the rt1020 in Zephyr version `2.7.3` `works` with Windows 10 on the above two builds and versions when connected directly to a usb host on the motherboard or a usb host via a usb hub.
-  a usb device built for the `mimxrt1064-evk` in Zephyr version `3.2.99` `works` with Windows 10 on the above two builds and versions when connected directly to a usb host on the motherboard or a usb host via a usb hub.
- a usb device built for the rt1020 in Zephyr version `3.2.99` `does not work` with a phone (android) via otg adapter 

Updated: now I'm testing and there are no problems, and I'm on the same commit and version of Windows. I have no explanation for what is happening.

**Comment by DerekSnell** [comment]
Hi @MikhailSiomin ,
I see your comment that things are magically working for you now.  I know that can be frustrating.

FWIW, my Windows is Windows 10 Enterprise 21H2 build `19044.2251`.  Best regards
